File: src/main/java/com/claudio_personagens/bookingcharactersapi/controllers/UserController.java
package com.claudio_personagens.bookingcharactersapi.controllers;

import com.claudio_personagens.bookingcharactersapi.domain.User;
import com.claudio_personagens.bookingcharactersapi.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(path = "users")
public class UserController {

    @Autowired
    private UserService userService;

    @PostMapping
    public User save(@RequestBody User user) {
        return userService.save(user);
    }

    @GetMapping
    public List<User> getUsers() {
        return userService.findAll();
    }
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/controllers/UserController.kt
package com.claudio_personagens.bookingcharactersapi.controllers

import com.claudio_personagens.bookingcharactersapi.security.domain.User
import com.claudio_personagens.bookingcharactersapi.services.UserService
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(UserController.URL)
class UserController (
    private val userService: UserService
) {

    companion object {
        @JvmStatic
        private val logger = LoggerFactory.getLogger(UserController::class.java)

        const val URL = "/users"
    }

    @PostMapping
    fun save(@RequestBody user: User) = userService.save(user)

    @GetMapping
    fun getlAll() = userService.findAll()
}